application/models/LabelSet.php

```diff
@@ -286,4 +287,14 @@ public function hasPermission($sPermission, $sCRUD = 'read', $iUserID = null)
         /* Finally : return specific one : always false if not  */
         return Permission::model()->hasPermission($this->lid, 'labelset', $sPermission, $sCRUD, $iUserID);
     }
+
+    /**
+     * Sanitize string for any attribute
+     * @param string $attribute to sanitize
+     * @return string sanitized attribute
+     */
+    public static function sanitizeAttribute($attribute)
+    {
+        return filter_var($attribute, @FILTER_SANITIZE_STRING);
```

Comment:
What's the `@` for in front of the constant? Never used it like that... 

Also, do we have any other way of sanitizing on ther models? I am realziing, `FILTER_SANITIZE_STRING` is deprectaed. So if using something else somwehere, better use that. right?

Reply:
> Also, do we have any other way of sanitizing on ther models?

No, else i use it. Near copy/paste from Token here.

Reply:
> FILTER_SANITIZE_STRING` is deprectaed.

If i understand right : it's FILTER_FLAG_NO_ENCODE_QUOTES deprecated.

> Encoding quotes can be disabled by setting FILTER_FLAG_NO_ENCODE_QUOTES. (Deprecated as of PHP 8.1.0, use [htmlspecialchars()](https://www.php.net/manual/en/function.htmlspecialchars.php) instead.)

Reply:
Look at the changelog section here.
https://www.php.net/manual/en/filter.filters.sanitize.php
![image](https://github.com/LimeSurvey/LimeSurvey/assets/4942495/d2d2f1ac-6409-4765-ae2e-72a4fd2592d2)

> If i understand right : it's FILTER_FLAG_NO_ENCODE_QUOTES deprecated.

>> Encoding quotes can be disabled by setting FILTER_FLAG_NO_ENCODE_QUOTES. (Deprecated as of PHP 8.1.0, use [htmlspecialchars()](https://www.php.net/manual/en/function.htmlspecialchars.php) instead.)

It is a bit confusing, but in changelog is clear.

Reply:
A good old preg_replace ?

Reply:
Or a `FILTER_SANITIZE_SPECIAL_CHARS` ?

Reply:
strip_tags ?

Reply:
hmm whatever is already used in other models for similar situations... :) sanitize _helper?

Reply:
> hmm whatever is already used in other models for similar situations... :) sanitize _helper?

There no global santizer validator, only xss, but here : i jyust don't want (care) of HTML, and needed for all user (admin included)

Found only 2 place

- https://github.com/LimeSurvey/LimeSurvey/blob/d01845f9895f4ada9abe31e0b2a5b772468babc5/application/models/Token.php#L322
- https://github.com/LimeSurvey/LimeSurvey/blob/d01845f9895f4ada9abe31e0b2a5b772468babc5/application/models/Token.php#L311